src/gamecontrol.cpp

```diff
@@ -234,54 +270,69 @@ void GameControl::QuitGame()
 }
 
 /** Initialize all game data structures for playing a new game. */
-void GameControl::NewLevel()
+void GameControl::InitializeLevel()
 {
 	Random::Initialize();
-	_scenario.SetDefaultScenario();  // \todo load a scenario.
-
-	/// \todo We blindly assume game data structures are all clean.
-	_world.SetWorldSize(20, 21);
-	_world.MakeFlatWorld(8);
-	_world.SetTileOwnerGlobally(OWN_NONE);
-	_world.SetTileOwnerRect(2, 2, 16, 15, OWN_PARK);
-	_world.SetTileOwnerRect(9, 1,  1,  1, OWN_PARK);
-	_world.SetTileOwnerRect(2, 18, 16, 2, OWN_FOR_SALE);
-
-	std::vector<const SceneryType*> park_entrance_types = _scenery.GetAllTypes(SCC_SCENARIO);
-	if (park_entrance_types.size() < 3) {
-		_world.SetTileOwnerRect(8, 0, 4, 2, OWN_PARK); // Allow building path to map edge in north west.
+
+	if (this->next_scenario != nullptr) {
+		_scenario = this->next_scenario->scenario;
+		_scenario.wrapper = this->next_scenario;
+		_scenario.name  = _language.GetSgText(this->next_scenario->name);
+		_scenario.descr = _language.GetSgText(this->next_scenario->descr);
 	} else {
-		/* Assemble a park entrance and some paths. This assumes that the entrance parts are the first three scenario scenery items loaded. */
-		_world.AddEdgesWithoutBorderFence(Point16(9, 0), EDGE_SE);
-		for (int i = 0; i < 3; i++) {
-			SceneryInstance *item = new SceneryInstance(park_entrance_types[i]);
-			item->orientation = 0;
-			item->vox_pos.x = 8 + i;
-			item->vox_pos.y = 1;
-			item->vox_pos.z = (i == 1 ? 12 : 8);
-			_scenery.AddItem(item);
-		}
-		for (int i = 0; i < 6; i++) {
-			BuildFlatPath(XYZPoint16(9, i, 8), PAT_CONCRETE, PAS_NORMAL_PATH, false, false);
+		/* When not loading a scenario, create some default objects. \todo Clean this mess up. */
+
+		_scenario.SetDefaultScenario();
```

Comment:
Don't know how you implemented this, but perhaps have a "default scenario" in the RCD files?
Then you don't have this case, and changes are not here but in the data of that scenario.